app/components/subscription/successful-subscription-modal.tsx

```diff
@@ -0,0 +1,55 @@
+import { useCallback } from "react";
+import { useNavigate, useSearchParams } from "@remix-run/react";
+import { AnimatePresence } from "framer-motion";
+import { Button } from "../shared/button";
+
+export default function SuccessfulSubscriptionModal() {
+  const [params] = useSearchParams();
+  const navigate = useNavigate();
+  const success = params.get("success") || false;
+  const handleBackdropClose = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      if (e.target !== e.currentTarget) return;
+      navigate("/settings/subscription");
```

Comment:
This works and is a really good way to do it in remix, but I tend to use navigate when the actual path changes. And searchParams, even tho they are part of the link they don't cause navigation. 
Thankfully remix has a super easy way to do it. the `useSearchParams` hook also has a setter.
`const [searchParams, setSearchParams] = useSearchParams()`

and then you can use setSearchParams to clean them up instead of using navigate. Both will work the same but this seems a better way as what we want to logically do is simply clear the params, not navigate from 1 url to another.